src/Microsoft.AspNet.Mvc.Razor/RazorViewEngine.cs

```diff
@@ -228,6 +228,29 @@ public virtual IEnumerable<string> AreaViewLocationFormats
             return new RazorPageResult(pageName, searchedLocations);
         }
 
+        private string GetNormalizedRouteValue(string key, ActionContext context)
+        {
+            if (context.ActionDescriptor.AttributeRouteInfo == null)
+            {
+                var match = context.ActionDescriptor.RouteConstraints.SingleOrDefault(r => r.RouteKey == key);
```

Comment:
Does this need to be `SingleOrDefault` as opposed to `FirstOrDefault`? `SingleOrDefault` will throw if you have multiple entries which match the constraint. If we want to explicitly deal with this scenario, we should throw a better exception.